conftest.py

```diff
@@ -329,6 +329,41 @@ def sr_disk_for_all_hosts(pytestconfig, request, host):
         logging.info(f">> Disk or block device {disk} is present and free on all pool members")
     yield candidates[0]
 
+@pytest.fixture(scope='session')
+def sr_disks_for_all_hosts(pytestconfig, request, host):
+    disks = pytestconfig.getoption("sr_disk")
+    assert len(disks) > 0, "This test requires at least one --sr-disk parameter"
+    # Fetch available disks on the master host
+    master_disks = host.available_disks()
+    assert len(master_disks) > 0, "A free disk device is required on the master host"
+
+    if "auto" not in disks:
+        # Validate that all specified disks exist on the master host
+        for disk in disks:
+            assert disk in master_disks, \
+                f"Disk or block device {disk} is either not present or already used on the master host"
+        master_disks = [disk for disk in disks if disk in master_disks]
```

Comment:
It would be more clear here to set `candidates` directly, and have the `candidates = list(master_disks)` be in an `else` statement (avoid overloading of `master_disks`)